File: Algorithm/Leetcode/[E]796. Rotate String.java
class Solution {
    public boolean rotateString(String s, String goal) {
        String s2 = s + s;
        if (s2.contains(goal) && s.length() == goal.length())
            return true;
        else
            return false;
            
    }
}


//s를 두번더해 s안에 goal이 있는지 확인하는 방법.
//String의 contains, length 메소드

File: Algorithm/Leetcode/[E]796. Rotate String.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        for i in range(len(s)):
            s = s[1:] + s[0]
            if s == goal:
                return True
            
        return False
    
# 간단한 문제 str을 쪼개는 방식으로 해당 값이 나오는지 확인했다
# 해당 방법보다 더 나은 방식으로 푸는 방법도 존재했다. 
# s + s 를해서 goal이 그 안에 있는지 확인하는방법